service/telemetry/internal/migration/v0.2.0.go

```diff
@@ -37,13 +37,12 @@ type logsConfigV020 struct {
 }
 
 func headersV02ToV03(in configv02.Headers) []config.NameStringValuePair {
-	headers := make([]config.NameStringValuePair, len(in))
-	idx := 0
+	headers := make([]config.NameStringValuePair, 0, len(in))
 	for k, v := range in {
-		headers[idx] = config.NameStringValuePair{
```

Comment:
There was another bug here that you never increased the idx.

Reply:
yeah that was the bug